visivo/models/model.py

```diff
@@ -1,19 +1,28 @@
 from typing import Optional
-from pydantic import Field
-from .base.base_model import BaseModel
+from pydantic import Field, validator
+from visivo.models.base.named_model import NamedModel
+from visivo.models.base.parent_model import ParentModel
+from visivo.models.target import DefaultTarget, Target
+from .base.base_model import generate_ref_field
 
 
-class Model(BaseModel):
+class Model(NamedModel, ParentModel):
     """
     Models are queries that return base data that is used in Traces
     """
 
-    name: Optional[str] = Field(
+    sql: str = Field(
         None,
-        description="Name of the model",
+        description="The sql used to generate your base data",
     )
 
-    sql: str = Field(
+    target: Optional[generate_ref_field(Target)] = Field(
```

Comment:
🥳 so it pretty much functions the same way that it did with the trace where if you don't specify it then it will use the default, that's great.